### PR TITLE
20201030 maxminapi

### PR DESCRIPTION
**Pull request by changlong-liu**
for issue https://github.com/Azure/autorest.az/issues/583

### CODE REVIEW

**Comment by qiaozha** [comment]
@changlong-liu  Could you also merge master code into your branch ? 

Also Can we use a common patten to deal with the resourceType minApi and maxApi https://github.com/Azure/autorest.az/blob/master/src/utils/helper.ts#L537-L544 

I just thought whether has min_api or max_api ResourceType in the generated code seems has not relations with whether it's commandGroup or command or method or methodParameter. The if else logic are pretty similar right?

File: src/test/scenarios/managed-network/configuration/readme.az.md [hunk 1]
```diff
@@ -58,6 +59,24 @@ cli:
             op: CreateOrUpdate
             param: properties
         poly-resource: true
+      - where:
+            group: managedNetworkPeeringPolicies
+            op: Delete
+        max-api: '2020-09-01'
+        min-api: '2019-09-01'
+        # resource-type: DATA_COMPUTE   # enable this after clicommon support it
```
Comment (qiaozha) [inline src/test/scenarios/managed-network/configuration/readme.az.md:67]: Does this mean clicommon haven't support it yet?

Comment (changlong-liu) [reply src/test/scenarios/managed-network/configuration/readme.az.md:67]: not support yet,  I have a PR here for it: https://github.com/Azure/autorest.clicommon/pull/101

**Review by qiaozha — APPROVED** [review]

**Pull request merged**